lib/reservation_manager.rb

```diff
@@ -0,0 +1,126 @@
+require_relative "reservation"
+
+class Reservation_manager
+  attr_reader :reservations, :pending_reservations_for_blocks
+
+  def initialize
+    @reservations = []
+    @pending_reservations_for_blocks = []
+  end
+
+  def all_rooms
+    @all_rooms = []
+    20.times do |number|
+      @all_rooms << number + 1
+    end
+    return @all_rooms
+  end
```

Comment:
Since the value for `@all_rooms` never changes (and never could change), it might make sense to set this value for `@all_rooms` just in the `initialize` method. Then, to access it, instead of making it an instance method, you could make an `attr_reader`, which will make an accessing (reading/getting) method for that instance variable. Therefore, I might say:

```ruby
  attr_reader :reservations, :pending_reservations_for_blocks, :all_rooms

  def initialize
    @reservations = []
    @pending_reservations_for_blocks = []
    @all_rooms = []
    20.times do |number|
      @all_rooms << number + 1
    end
  end
```